packages/salesforce-adapter/src/filters/changed_at_singleton.ts

```diff
@@ -59,6 +63,13 @@ const getChangedAtSingletonInstance = async (
   return changedAtSingleton ?? createEmptyChangedAtSingletonInstance()
 }
 
+const dateStringOfMostRecentlyChangedInstance = (instances: InstanceElement[]): string => (
+  _(instances)
+    .map(instance => instance.annotations[CORE_ANNOTATIONS.CHANGED_AT])
+    .filter(changedAt => changedAt !== undefined)
+    .maxBy((changedAt: string) => new Date(changedAt).getTime())
+)
```

Comment:
What if no element has `_changed_at` value? The return type is incorrect and since we iterate any values from `L68` TS didn't catch that.

```suggestion
const dateStringOfMostRecentlyChangedInstance = (instances: InstanceElement[]): string | undefined => (
  _(instances)
    .map(instance => instance.annotations[CORE_ANNOTATIONS.CHANGED_AT])
    .filter(_.isString)
    .filter(changedAt => changedAt !== '')
    .maxBy(changedAt => new Date(changedAt).getTime())
)
```

Please also note we should make sure we don't deal with empty string values (which in some flows of the adapter are what Salesforce returns, specifically for Metadata Instances, which is not the use-case here, but since the received argument is instances we should be bulletproof there)

Consider exporting to a utility `isValidChangedAt(value: unknown): value is string`